src/plugins/telemetry/public/plugin.ts

```diff
@@ -136,7 +137,9 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
       currentKibanaVersion,
     });
 
-    this.telemetrySender = new TelemetrySender(this.telemetryService);
+    this.telemetrySender = new TelemetrySender(this.telemetryService, async () => {
+      await this.refreshConfig();
```

Comment:
How expensive would this additional async task potentially get in terms of performance? We've already had  to implement a back-off mechanism for failed send-requests and I'm a little concerned that adding too much before each send attempt might make the situation worse.

Reply:
That's a very valid concern! Good catch! To be fair, this only requires one SO get-by-id request. IMO, it's a very cheap one. However, I agree that we should try to minimize the number of times we request it. I'll see if I can narrow down the number of times we request it.